src/components/listing-page/Overview/index.jsx

```diff
@@ -24,8 +27,20 @@ import {
  * @return {JSX.Element}
  */
 function Overview({ listing }) {
+  const isOwner = useRef(null);
   const { schedule, id } = listing;
   const { setShowModal } = useShowModal();
+  const { id: formId } = useParams();
+  const { userInfo } = useUser();
+
+  const getData = async () => {
+    const response = await getListingsById(formId);
+    isOwner.current = response.agentId === userInfo.id;
+  };
```

Comment:
```suggestion
  };
  
```
why are you using a ref instead of a state?